src/charm.py

```diff
@@ -140,6 +150,24 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
         # results is a dict and set_results expects _SerializedData
         event.set_results(results)  # type: ignore[arg-type]
 
+    def _on_database_created(self, event: DatabaseCreatedEvent) -> None:
+        """Handle database created.
+
+        Args:
+            event: Event triggering the database created handler.
+        """
+        if not self.unit.is_leader():
+            event.defer()
+            return
+
+        self.model.unit.status = ops.MaintenanceStatus("Preparing the database")
+        try:
+            self._database.prepare_database()
+            self._on_config_changed(event)
```

Comment:
I think the business logic should be moved to a dedicated method that both hook method will call instead of calling the hook method directly

Reply:
Hm...When there is a database relation, the configuration file should be updated considering the database information like a configuration change would do. Unless I change to something like:

```
def update_config...

def _config_changed:
    change_config()

def _on_database_created:
    ...
    update_config()
   ...
```
Is that what you mean?

Reply:
Yes I suspect that is what Gregory means. Event handler function should only be used to handle events and the code suggested above is a way of sharing the business logic between events

Reply:
Thanks, changed.